### PR TITLE
refactor(Worklets): Move shareable creation to Worklets

### PR DESCRIPTION
**Pull request by tjzel**
## Summary

Moving `makeShareableClone` function from `NativeReanimatedModule` to `NativeWorkletsModule`.

## Test plan

CI

### CODE REVIEW

**Review by piaskowyk — CHANGES_REQUESTED** [review]
I like the rest of code 🫡

File: packages/react-native-reanimated/src/commonTypes.ts [hunk 1]
```diff
@@ -7,7 +7,13 @@ import type {
 } from 'react-native';
 import type { WorkletRuntime } from './runtimes';
 
-export interface IWorkletsModule {}
+export interface IWorkletsModule {
```
[the remark this answers is not in the document]
Comment (tjzel) [reply packages/react-native-reanimated/src/commonTypes.ts:10]: These are not the same. `IWorkletsModule` specifies the interface for the `WorkletsModule` class - which uses `__workletsModuleProxy` as it's member.

[the remark this answers is not in the document]
Comment (tjzel) [reply packages/react-native-reanimated/src/commonTypes.ts:10]: But the duplication can be reduced, let me think about it.

[the remark this answers is not in the document]
Comment (tjzel) [reply packages/react-native-reanimated/src/commonTypes.ts]: Done in [fe37091](https://github.com/software-mansion/react-native-reanimated/pull/6576/commits/fe37091af1eeacfae089d810c8219a39335a4cca)

**Review by piaskowyk — APPROVED** [review]

**Pull request merged**